### PR TITLE
Ports rebindable movement keys from /tg/

### LINKED ISSUE #3633: Ability to keybind movement keys to other keys.
As a player with a non qwerty keyboard, playing bee is a downright nightmare of having to switch between a french layout and english layout mid game. It leads to a lot of confusion, annoyance and anger, for no real reason as a keybind menu is already coded in - but for some reason, I cannot change the keybinds of my movement keys.

I've been playing bee for a month, and despite wanting really strongly to stay, this one missing feature is legitimately ruining my experience to a point where i just want to stop playing sometimes.

For the love of all that is holy, please. Let us keybind all possible keys.

### PR DESCRIPTION
**Pull request by super12pl**
## About The Pull Request
Ports https://github.com/tgstation/tgstation/pull/47626
Adds ability to rebind default movement keys (WSAD) in game preferences.

I am not sure about quality of the port but hey it works.

## Why It's Good For The Game
Makes the game *actually playable* for people with non-qwerty keyboards, fixes #3633, also more editable hotkeys good.

## Changelog
:cl: Couls
add: Rebindable movement keys
/:cl:

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -968,6 +968,21 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 /client/proc/rescale_view(change, min, max)
 	view_size.setTo(clamp(change, min, max), clamp(change, min, max))
 
+/client/proc/update_movement_keys()
+	if(prefs && prefs.key_bindings)
```
Comment (utilbird) [inline code/modules/client/client_procs.dm]:
```suggestion
	if(prefs?.key_bindings)
```

**Comment by francinum** [comment]
Testmerged for RID 29867, resulted in nonfunctional movement keys.

**Review by francinum — CHANGES_REQUESTED** [review]
Nonfunctional upon testmerge.

No relevant runtimes.

**Comment by francinum** [comment]
Additional aside to the review:

* Set sane default hotkeys **and force them with a savefile version burn**
* Arrowkeys *MUST* always be valid movement keys.

**Comment by super12pl** [comment]
¯\_(ツ)_/¯

**Pull request closed without merging**